File: 0199-binary-tree-right-side-view/0199-binary-tree-right-side-view.java
/**
 * Definition for a binary tree node.
 * public class TreeNode {
 *     int val;
 *     TreeNode left;
 *     TreeNode right;
 *     TreeNode() {}
 *     TreeNode(int val) { this.val = val; }
 *     TreeNode(int val, TreeNode left, TreeNode right) {
 *         this.val = val;
 *         this.left = left;
 *         this.right = right;
 *     }
 * }
 */
class Solution {
    public List<Integer> rightSideView(TreeNode root) {
        List<Integer> ans = new ArrayList<>();
        sideView(root, 0, ans);
        return ans;
    }
    void sideView(TreeNode root, int level, List<Integer> ans){
        if(root==null) return;
        if(ans.size() == level) ans.add(root.val);
        sideView(root.right, level+1, ans);
        sideView(root.left, level+1, ans);
    }
}

File: 0199-binary-tree-right-side-view/0199-binary-tree-right-side-view.py
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, val=0, left=None, right=None):
#         self.val = val
#         self.left = left
#         self.right = right
class Solution:
    def rightSideView(self, root: Optional[TreeNode]) -> List[int]:
        ans = []
        self.sideView(root, 0, ans)
        return ans
    def sideView(self, root, level, ans):
        if not root: return;
        if len(ans) == level: ans.append(root.val)
        self.sideView(root.right, level+1, ans)
        self.sideView(root.left, level+1, ans)